cypress-io/cypress
# fix: Recovered Error Handling and Session Recovered Errors

- Closes #21398
- Closes #24205
- Closes #24593

There PR has a few set of changes going on:
- Correct `cy.session()` error messaging for both created and restored/created with test coverage
	- updated messaging
	- proper code frames
	- handles the following failed validation scenarios: yield false, rejects, resolve false, thrown cy command, thrown err
	- REMOVED support for return false
- Fix Session Instrument Panel Tag color
- Enhance `cy.session()` console props output to include sessionStorage details and add step information to help with debuggin
- Fix verifying upcoming assertions log handling (and associated tests) - this was need to prevent duplicate error logging in recreated sessions
- Update command queue to be recursive to allow queue failures to always be collected when `onQueueFailed` (previously `onCommandFailed`) to ensure all failures can be collected on a 2nd session validation failure and/or when multiple session commands are used in a test.
- Correctly end log groups when `on:fail` handlers to make additional assertions visible 

### User facing changelog
TO DO

### Steps to test
It would be solid to pull this branch down and run various tests yourself to verify changes to logs and to ensure we don't hanging logs and that things appear to be correct. 

### How has the user experience changed?
Will comment screenshots here & throughout the code. 

### PR Tasks
- [x] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? https://github.com/cypress-io/cypress-documentation/pull/4842
